x-pack/plugins/alerting/server/rules_client/lib/get_alert_from_raw.ts

```diff
@@ -158,7 +159,12 @@ export function getPartialRuleFromRaw<Params extends RuleTypeParams>(
 
   if (omitGeneratedValues) {
     if (rule.actions) {
-      rule.actions = rule.actions.map((ruleAction) => omit(ruleAction, 'alertsFilter.query.dsl'));
+      rule.actions = rule.actions.map((ruleAction) => {
```

Comment:
Note that we're deprecating `getAlertFromRaw` in favor of `transformRuleDomainToRuleAttributes`, so let's make sure these changes are reflected there as well

Reply:
I think that analog of old `getAlertFromRaw` is a new `transformRuleAttributesToRuleDomain`. I think I've already added the smae type guard there as well: `x-pack/plugins/alerting/server/application/rule/transforms/transform_rule_attributes_to_rule_domain.ts` line 157.
Isn't what you meant?

Reply:
Thanks for the feedback! This PR is the first of many. I modified `transformRuleDomainToRuleAttributes` in https://github.com/elastic/kibana/pull/167884/files#diff-a4ecdf971673064aa6dc4ab5620e7b128efcc27b3e77804a3753d2839eaa7ba2L153. To keep the size small I choose to not do all the changes. My first [PR](https://github.com/elastic/kibana/pull/167883) tried to accommodate everything, and it got too big. We close it in favor of smaller PRs that deliberately will not pass the CI and will get merged in a feature branch.

Reply:
@cnasikas I did not recognise your intention from the beginning and started to fix all types :-)
Now I see that you made some of this changes in follow up PRs. So I've deleted my commit for this PR.